### ISSUE TITLE
Use with RTK Query - unable to assign request headers (specifically `Content-Type` and `rid`)

### ISSUE THREAD

**Issue by anmilleriii** [supertokens/supertokens-website#116]
Edit: this does pertain to ST's implementation of a `fetch` wrapper.

~~Possibly out of scope of ST since this is for use with a third-party library (Redux Toolkit (RTK) Query), please close if so.~~

As described in this RTK issue: https://github.com/reduxjs/redux-toolkit/issues/2205#issuecomment-1094141663:

The expected behavior is the headers should be included in the request if included in any of the ways described (using `prepareHeaders`, assigning object literal to `headers`, or assigning the header within the `header` key in an endpoint's `query`).

The observed behavior headers are not able to be assigned (including `rid` and `Content-Type` to HTTP requests made with `fetchBaseQuery` (see https://redux-toolkit.js.org/rtk-query/api/fetchBaseQuery#using-fetchbasequery) w/SuperTokens. This results in the request body not being parsed/sent correctly.

Thought I would open here since RTK Query is gaining traction, thanks!

**Comment by phryneas** [comment]
This is not specific to RTK Query - it seems that `supertokens` creates a wrapper around `fetch` that does not comply with the standard.

The first argument to ]`fetch`](https://fetch.spec.whatwg.org/#fetch-method) has to be a [RequestInfo](https://fetch.spec.whatwg.org/#requestinfo) which is either a `string`, or a `Request` class instance. 

The supertokens wrapper only accepts a string, RTK Query calls it with a `Request` object.

**Comment by rishabhpoddar** [comment]
Oh I see! We shall fix this right away. Thanks for the issue.

**Comment by rishabhpoddar** [comment]
On seeing the code again, our `fetch` wrapper's first param is indeed a `RequestInfo`: https://github.com/supertokens/supertokens-website/blob/master/lib/ts/recipeImplementation.ts#L9.

But I will still see why this is not working

**Comment by phryneas** [comment]
It seems to come from you calling 
```js
                        return originalFetch(url, {
                            ...config
                        });
```

Reading the spec, that will lead to a call to `new Request(url, {...config})` internally.

Assuming that `url` in that case is a `Request` object that already has headers, going by the "new Request(input, init) constructor steps" described in the standard, any `headers` in `config` would overwrite the existing `headers` in `url` completely instead of merging them.

**Comment by rishabhpoddar** [comment]
hey @anmilleriii we are in progress of making a fix. In the meantime, can you please try using supertokens-website from [this branch](https://github.com/supertokens/supertokens-website/tree/fix_fetch_headers) and see if it solves the issue? It will help us make sure we are on the right track.

You can use that version by running the command:
### if using supertokens-website
```bash
npm i git+https://github.com/supertokens/supertokens-website#fix_fetch_headers
```

### else if using supertokens-auth-react
```bash
rm -rf node_modules/supertokens-website # if this fails, ignore the failure
rm -rf node_modules/supertokens-auth-react/node_modules/supertokens-website # if this fails, ignore the failure
npm i git+https://github.com/supertokens/supertokens-website#fix_fetch_headers
```

**Comment by anmilleriii** [comment]
Hey @rishabhpoddar, sorry for the delay!

This fix worked (yay!)

In the process of debugging this, I realize I had a significant misunderstanding of the role of `supertokens-auth-react` versus `supertokens-website`.

I thought I should be using `supertokens-auth-react` since I have a React project - however, I am using a custom UI and directly integrating with the OpenAPI endpoints, so I don't need `supertokens-auth-react` at all (is this correct?).

Here is a reproduction of the (now functional with the `fix_fetch_headers` branch) fix. 

https://github.com/anmilleriii/rtk-query-cors-supertokens-reproduction

You can validate that fixed the issue by switching to any different version of `supertokens-website`, which will result in a `{"message":"Missing input param: formFields"}` response due to the missing headers (including `Content-Type`).

I was also able to trivially work around this issue by forcing my ST node backend to parse the response regardless of missing `Content-Type` header. However this fix doesn't treat the root cause/has other issues (for instance not receiving all the headers it needs):

```javascript
/**
 * @see https://stackoverflow.com/questions/17227881/how-to-parse-http-request-with-a-missing-content-type-in-express-nodejs-by-ass
 */
app.use(
  express.json({
    inflate: true,
    strict: false,
    type: () => {
      return true;
    },
  })
);
```

Finally, I noted that this issue did not affect requests made outside of browser (e.g., Postman), since the headers that weren't sent were not required to correctly handle the response if CORS isn't applicable.

Note I did not test the `supertokens-auth-react` update, but would assume that would work since it just wraps `supertokens-website`.

Thanks for the assistance!

**Comment by rishabhpoddar** [comment]
> however, I am using a custom UI and directly integrating with the OpenAPI endpoints, so I don't need supertokens-auth-react at all (is this correct?).

Yes. It is correct.

Thanks for the testing it out @anmilleriii 

And thanks for pointing out the fix @phryneas

**Comment by rishabhpoddar** [comment]
Released in version 10.0.10 of the supertokens-website SDK
